projects/net-snmp/parse_octet_hint_fuzzer.c

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <net-snmp/net-snmp-config.h>
+#include <net-snmp/net-snmp-includes.h>
+#include <assert.h>
+#include <stddef.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <string.h>
+
+int LLVMFuzzerInitialize(int *argc, char ***argv) {
+    if (getenv("NETSNMP_DEBUGGING") != NULL) {
+        /*
+         * Turn on all debugging, to help understand what
+         * bits of the parser are running.
+         */
+        snmp_enable_stderrlog();
+        snmp_set_do_debugging(1);
+        debug_register_tokens("");
+    }
+    return 0;
+}
+
+int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size) {
+    char *hint, *value;
+    size_t value_start, new_val_len;
+    unsigned char *new_val;
+
+    if (size == 0 || data[size - 1] != '\0')
```

Comment:
So this will only pass in inputs where the fuzzer gets the last byte right in terms of a NULL-terminated string. However, why not wrap the data in a NULL-terminated string itself, thus ensuring every fuzz input can be passed? Similar to how we do it here: https://github.com/google/oss-fuzz/blob/4a13e59f3a09cece14001ed747c08ef06a6fb126/projects/net-snmp/snmp_octet_fuzzer.c#L48